ui/lib/core/addon/components/search-select-with-modal.js

```diff
@@ -13,137 +12,148 @@ import { filterOptions, defaultMatcher } from 'ember-power-select/utils/group-ut
+        searchText: matchingOption ? matchingOption.searchText : option,
+      };
+    });
   }
 
-  @action
-  async fetchOptions() {
-    try {
-      const queryOptions = {};
-      const options = await this.store.query(this.args.model, queryOptions);
-      this.formatOptions(options);
-    } catch (err) {
-      if (err.httpStatus === 404) {
-        if (!this.allOptions) {
-          // If the call failed but the resource has items
-          // from a different namespace, this allows the
-          // selected items to display
-          this.allOptions = [];
+  @task
+  *fetchOptions() {
+    this.dropdownOptions = []; // reset dropdown anytime we re-fetch
+    if (!this.args.models) {
+      return;
+    }
+
+    for (const modelType of this.args.models) {
+      try {
+        const queryParams = {};
+        // fetch options from the store
+        let options = yield this.store.query(modelType, queryParams);
+        if (this.args.excludeOptions) {
+          options = options.filter((o) => !this.args.excludeOptions.includes(o.id));
         }
-        return;
-      }
-      if (err.httpStatus === 403) {
-        this.shouldUseFallback = true;
-        return;
+        // add to dropdown options
+        this.dropdownOptions = [...this.dropdownOptions, ...this.addSearchText(options)];
+      } catch (err) {
+        if (err.httpStatus === 404) {
+          // continue to query other models even if one 404s
+          // and so selectedOptions will be set after for loop
+          continue;
+        }
+        if (err.httpStatus === 403) {
+          if (this.dropdownOptions && this.args.models.length > 1) continue;
+          this.shouldUseFallback = true;
+          return;
+        }
+        throw err;
       }
-      throw err;
     }
-  }
-  formatOptions(options) {
-    options = options.toArray();
-    if (this.excludeOptions) {
-      options = options.filter((o) => !this.excludeOptions.includes(o.id));
-    }
-    options = options.map((option) => {
-      option.searchText = `${option.name} ${option.id}`;
-      return option;
-    });
 
-    if (this.selectedOptions.length > 0) {
-      this.selectedOptions = this.selectedOptions.map((option) => {
-        const matchingOption = options.findBy('id', option);
-        options.removeObject(matchingOption);
-        return {
-          id: option,
-          name: matchingOption ? matchingOption.name : option,
-          searchText: matchingOption ? matchingOption.searchText : option,
-        };
-      });
-    }
-    this.allOptions = options;
+    // after all models are queried, set selectedOptions and remove matches from dropdown list
+    this.selectedOptions = this.args.inputValue
+      ? this.formatInputAndUpdateDropdown(this.args.inputValue)
+      : [];
   }
 
+  @action
   handleChange() {
     if (this.selectedOptions.length && typeof this.selectedOptions.firstObject === 'object') {
-      if (this.passObject) {
```

Comment:
I understand we might not need this in the modal version, but at some point we had to be able know in the parent whether the user picked a new item, so that we could show the form inline. Maybe in a future where the two components are merged, we can *only* show forms in a modal? 🤩